Lib/test/datetimetester.py

```diff
@@ -2587,15 +2602,22 @@ def test_fromisoformat_separators(self):
             ' ', 'T', '\u007f',     # 1-bit widths
             '\u0080', 'ʁ',          # 2-bit widths
             'ᛇ', '時',               # 3-bit widths
-            '🐍'                     # 4-bit widths
+            '🐍',                    # 4-bit widths
+            '\ud800',               # bpo-34454
         ]
 
         for sep in separators:
             dt = self.theclass(2018, 1, 31, 23, 59, 47, 124789)
             dtstr = dt.isoformat(sep=sep)
 
             with self.subTest(dtstr=dtstr):
-                dt_rt = self.theclass.fromisoformat(dtstr)
+                try:
```

Comment:
I don't think this try/catch is the right solution here. We should fix the actual problem.